library/Fission/Random.hs

```diff
@@ -1,42 +1,32 @@
 -- | Random values
 module Fission.Random
-  ( byteString
-  , text
+  ( alphaNum
+  , alphaNumSymbol
   ) where
 
 import qualified Data.ByteString.Random as BS
-import           Data.Word8
 import qualified RIO.ByteString         as BS
+import qualified Fission.Internal.URL   as URL
+import           Data.Text              as T
+import           Data.Word8
+import           GHC.Natural
 
-import Fission.Internal.Bool (anyX)
 import Fission.Prelude
 
--- | Generate random 'Text'
-text :: Natural -> IO Text
-text amount = decodeUtf8Lenient <$> byteString amount
+-- | Generate random AlphaNumeric 'Text' with a given length
+alphaNum :: Natural -> IO Text
+alphaNum len = decodeUtf8Lenient . BS.filter isAsciiAlphaNum <$> bsRandomLength len
 
--- | Generate a random 'ByteString'
-byteString :: Natural -> IO ByteString
-byteString amount = BS.filter isURL <$> BS.random amount
+-- | Generate random AlphaNumericSymbol 'Text' with a given length
+alphaNumSymbol :: Natural -> IO Text
+alphaNumSymbol len = decodeUtf8Lenient . BS.filter URL.isValidURLCharacter <$> bsRandomLength len
 
--- | Check that a byte represents a valid URL character
-isURL :: Word8 -> Bool
-isURL w = isAsciiUpper w
-        || isAsciiLower w
-        || isDigit w
-        || anyX urlSpecials w
+-- | Generate random 'ByteString' with a given length
+bsRandomLength :: Natural -> IO ByteString
+bsRandomLength len = BS.take (naturalToInt len) <$> BS.random (len * 100)
```

Comment:
Why `* 100`? Why not just pas that up to the caller?

Reply:
Anyhow, now that we're all pipey:

```haskell
len
  |> (* 100)
  |> BS.random
  |> fmap (BS.take toTake)
where
  toTake = toIntegral len
```

Reply:
Oooh! I prefer the pipes 🚰 😍